src/internal/m365/collection/drive/export.go

```diff
@@ -13,83 +13,61 @@ import (
 	"github.com/alcionai/corso/src/pkg/fault"
 )
 
-var _ export.Collection = &ExportCollection{}
-
-// ExportCollection is the implementation of export.ExportCollection for OneDrive
-type ExportCollection struct {
-	// baseDir contains the path of the collection
-	baseDir string
-
-	// backingCollection is the restore collection from which we will
-	// create the export collection.
-	backingCollection data.RestoreCollection
-
-	// backupVersion is the backupVersion of the backup this collection was part
-	// of. This is required to figure out how to get the name of the
-	// item.
-	backupVersion int
-}
-
 func NewExportCollection(
 	baseDir string,
-	backingCollection data.RestoreCollection,
+	backingCollection []data.RestoreCollection,
 	backupVersion int,
-) ExportCollection {
-	return ExportCollection{
-		baseDir:           baseDir,
-		backingCollection: backingCollection,
-		backupVersion:     backupVersion,
+) export.Collectioner {
+	return export.BaseCollection{
+		BaseDir:           baseDir,
+		BackingCollection: backingCollection,
+		BackupVersion:     backupVersion,
+		Stream:            streamItems,
 	}
 }
 
-func (ec ExportCollection) BasePath() string {
-	return ec.baseDir
-}
-
-func (ec ExportCollection) Items(ctx context.Context) <-chan export.Item {
-	ch := make(chan export.Item)
-	go items(ctx, ec, ch)
-
-	return ch
-}
-
-// items converts items in backing collection to export items
-func items(ctx context.Context, ec ExportCollection, ch chan<- export.Item) {
+// streamItems streams the streamItems in the backingCollection into the export stream chan
+func streamItems(
+	ctx context.Context,
+	drc []data.RestoreCollection,
+	backupVersion int,
+	ch chan<- export.Item,
+) {
 	defer close(ch)
 
 	errs := fault.New(false)
 
-	for item := range ec.backingCollection.Items(ctx, errs) {
-		itemUUID := item.ID()
-		if isMetadataFile(itemUUID, ec.backupVersion) {
-			continue
-		}
-
-		name, err := getItemName(ctx, itemUUID, ec.backupVersion, ec.backingCollection)
-
-		ch <- export.Item{
-			ID: itemUUID,
-			Data: export.ItemData{
-				Name: name,
-				Body: item.ToReader(),
-			},
-			Error: err,
+	for _, rc := range drc {
```

Comment:
Although it can be multiple collections, as of now I don't see a case where anything drive/file related having multiple collections. It would be good to have a comment here that we are only expecting one collection in this.

Reply:
Eh, that seems like a conflict of interest.  Either we need the slice or we don't.  But keeping it in and saying we don't want it feels weird.  I'm going to keep it in for now, and we can revisit if needed.